File: app/src/main/java/com/gersseba/garden/i18n/SettingsDataStore.java
package com.gersseba.garden.i18n;

/**
 * Java-friendly settings store interface for the app language.
 * Implementations should avoid blocking the main thread for writes.
 */
public interface SettingsDataStore {
	/**
	 * Returns the saved locale as a BCP-47 language tag or null if not set.
	 * Implementations should return quickly (SharedPreferences.getString is acceptable).
	 */
	String getSavedLocale();

	/**
	 * Persists the locale as a BCP-47 language tag. Implementations should use
	 * an asynchronous apply() or off-main-thread write so callers are not blocked.
	 */
	void saveLocale(String languageTag);
}

File: app/src/main/java/com/gersseba/garden/i18n/SettingsDataStore.kt
package com.gersseba.garden.i18n

/**
 * Minimal interface representing a persisted settings store for locale.
 * Implementations may use DataStore, SharedPreferences, or test fakes.
 *
 * Exposes suspend functions to avoid blocking callers.
 */
interface SettingsDataStore {
    /**
     * Returns the saved locale as a BCP-47 language tag or null if not set.
     */
    suspend fun getSavedLocale(): String?

    /**
     * Persists the locale as a BCP-47 language tag.
     */
    suspend fun saveLocale(languageTag: String)
}


